sycl/include/sycl/handler.hpp

```diff
@@ -438,14 +438,22 @@ class __SYCL_EXPORT handler {
   ///        is null if no secondary queue is associated with the submission.
   /// \param CallerNeedsEvent indicates if the event resulting from this handler
   ///        is needed by the caller.
+#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
+  // TODO: This function is not used anymore, remove it in the next
+  // ABI-breaking window.
   handler(std::shared_ptr<detail::queue_impl> Queue,
           std::shared_ptr<detail::queue_impl> PrimaryQueue,
           std::shared_ptr<detail::queue_impl> SecondaryQueue,
           bool CallerNeedsEvent);
+#endif
+  __SYCL_DLL_LOCAL handler(std::shared_ptr<detail::queue_impl> Queue,
```

Comment:
Why is the `Queue` still `std::shared_ptr`, but the `PrimaryQueue` and `SecondaryQueue` are raw pointers?

Reply:
To keep ABI stable. Queue is saved in `sycl::handler`, while `PrimaryQueue` and `SecondaryQueue` are go to `handler_impl`. I'm addressing 1st item in the ABI-breaking PR.

Reply:
I think it was just due to time constraints during the last ABI breaking window. I see no reason not to move more fields into handler_impl. 

It would probably be cleaner to move more of handler to handler_impl during the ABI breaking window itself though, rather than putting it under preview.